src/selection.ts

```diff
@@ -0,0 +1,62 @@
+export enum SelectionTypes {
```

Comment:
> avoid hardcoded constants

If you use string literal to declare types, hardcoding constant no longer cause misspelling problem because if you misspell the constants, the build won't pass.   Plus, with string literals, you already got nice autocompletion of possible values with a proper typescript editor (e.g., VSCode).  

Thus, I think that sticking with a) format above makes thing simpler since it reduces number of imports we have to do for each file without causing typo bugs. 

> For my own edification, could you explain why we've moved away from the enums? 

Enum suppose to be numbers, not string in Typescript.  Using as `any` was a hack that works okay in the past.  However, it prevents us from turning on `noImplictyAny` flag as request by external users of our project in #1649.   There might be also other reasons, but I can't remember one now.  Maybe @domoritz can add to this.